tests/unit/test_charm.py

```diff
@@ -730,3 +749,219 @@ def test_when_remove_then_external_service_is_deleted(self, patch_client):
             name=f"{self.harness.charm.app.name}-external",
             namespace=self.namespace,
         )
+
+    def test_given_default_config_when_network_attachment_definitions_from_config_is_called_then_no_interface_mtu_specified_in_nad(  # noqa: E501
+        self,
+    ):
+        self.harness.set_leader(is_leader=True)
+        self.harness.update_config(
+            key_values={
+                "access-ip": "192.168.252.3/24",
+                "access-gateway-ip": ACCESS_GW_IP,
+                "gnb-subnet": GNB_SUBNET,
+                "core-ip": CORE_IP,
+                "core-gateway-ip": CORE_GW_IP,
+            }
+        )
+        nads = self.harness.charm._network_attachment_definitions_from_config()
+        for nad in nads:
+            config = json.loads(nad.spec["config"])
+            self.assertNotIn("mtu", config)
+
+    def test_given_default_config_with_interfaces_mtu_sizes_when_network_attachment_definitions_from_config_is_called_then_mtu_sizes_specified_in_nad(  # noqa: E501
+        self,
+    ):
+        self.harness.set_leader(is_leader=True)
+        self.harness.update_config(
+            key_values={
+                "access-ip": "192.168.252.3/24",
+                "access-gateway-ip": ACCESS_GW_IP,
+                "access-interface-mtu-size": VALID_MTU_SIZE_1,
+                "gnb-subnet": GNB_SUBNET,
+                "core-ip": CORE_IP,
+                "core-gateway-ip": CORE_GW_IP,
+                "core-interface-mtu-size": VALID_MTU_SIZE_1,
+            }
+        )
+        nads = self.harness.charm._network_attachment_definitions_from_config()
+        for nad in nads:
+            config = json.loads(nad.spec["config"])
+            self.assertEqual(config["mtu"], 65535)
+
+    @patch(
+        f"{MULTUS_LIBRARY_PATH}.KubernetesMultusCharmLib._configure_network_attachment_definitions",  # noqa: E501
```

Comment:
We should not patch the lib private methods in our charm tests. There is no guarantee that this will remain the same. In general, we should only patch public method calls

Reply:
Done